src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -420,6 +439,48 @@ def _process_withdrawals(
             # End of pagination details
             pass
 
+    def __safe_api_call(
+        self,
+        function: Callable[..., Iterable[Dict[str, Union[str, float]]]],
+        params: Dict[str, Any],
+    ) -> Iterable[Dict[str, Union[str, float]]]:
+
+        results: Iterable[Dict[str, Union[str, float]]]
+        request_count: int = 0
+
+        # Most exceptions are caused by request limits of the underlying APIs
+        while request_count < 9:
+            try:
+                if "code" in params:
+                    results = function(
+                        code=params["code"],
+                        since=params["since"],
+                        limit=params["limit"],
+                        params=params["params"],
+                    )
```

Comment:
You should be able to use dict unpacking here:
```suggestion
                    results = function(**params)
```

Reply:
That's convenient. Learn something new every day. Thanks.